akka-stream/src/main/scala/akka/stream/ActorMaterializer.scala

```diff
@@ -129,12 +129,11 @@ object ActorMaterializer {
    * INTERNAL API
    */
   private[akka] def downcast(materializer: Materializer): ActorMaterializer =
-    materializer match {
+    materializer match { //FIXME this method is going to cause trouble for other Materializer implementations
```

Comment:
The problem is that you can only provide a single Materializer, and even if that were carrying an ActorMaterializer this piece of logic would fail. Needs reengineering IMO.